pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -169,9 +164,7 @@ def init_dist_connection(self, global_rank: int, world_size: int) -> None:
     def set_world_ranks(self, process_idx: int = 0) -> None:
         pass
 
-    def new_process(self, trainer: "pl.Trainer", mp_queue: SimpleQueue) -> None:
-        self.mp_queue = mp_queue
-
+    def new_process(self, trainer: "pl.Trainer") -> Optional[Tuple[Optional[str], Optional[str], Any, "_ExtraQueue"]]:
```

Comment:
In ddp (non-spawning plugins), we only need pass trainer to strategy for setup() which calls init_optimizer. After we clean up TrainerOptimizersMixin, most plugin doesn't need trainer anymore. 
Do you think it's possible to spawn individual functions instead of entire trainer for spawning in the furture?

Reply:
In step 4 and 5 of  #10059 we will automatically get rid of some of these trainer references yes. There are some occurrences in the plugins where we need to check things like `trainer.stage == FITTING` but these trainer instances can also be accessed through model.trainer if people think this is better.